### PR TITLE
[ebpf] decouple ebpf-telemetry from tracer

### PR DESCRIPTION
**Pull request by val06**
### What does this PR do?

https://datadoghq.atlassian.net/browse/EBPF-525

Decouple ebpf telemetry pkg from network's tracer. 
Move the collector registration to system-probe's initialization

### Motivation

Make the ebpf errors telemetry a truly generic package, which can be used by ANY ebpf program (e.g: CWS programs).

### Additional Notes

we should move the integration test from the TracerSuite into ebpf/telemetry and replace the tracer's ebpf program with dummy ebpf program to trigger the ebpf telemetry. This will be addressed in a separate [PR](https://github.com/DataDog/datadog-agent/pull/26849)

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

File: pkg/ebpf/telemetry/errors_telemetry.go [hunk 1]
```diff
@@ -80,17 +86,15 @@ func (e *ebpfTelemetry) Unlock() {
 }
 
 func (e *ebpfTelemetry) setup(opts *manager.Options) {
-	if (e.mapErrMap != nil) || (e.helperErrMap != nil) {
-		if opts.MapEditors == nil {
-			opts.MapEditors = make(map[string]*ebpf.Map)
-		}
+	if (e.mapErrMap != nil || e.helperErrMap != nil) && opts.MapEditors != nil {
```
Comment (brycekahle) [inline pkg/ebpf/telemetry/errors_telemetry.go:89]:
```suggestion
	if (e.mapErrMap != nil || e.helperErrMap != nil) && opts.MapEditors == nil {
```

File: cmd/system-probe/subcommands/run/command.go [hunk 2]
```diff
@@ -341,6 +341,10 @@ func startSystemProbe(log log.Component, statsd compstatsd.Component, telemetry
 			if lcc := ddebpf.NewLockContentionCollector(); lcc != nil {
 				telemetry.RegisterCollector(lcc)
 			}
+			telemetry.RegisterCollector(ddebpf.NewLockContentionCollector())
+			if ec := ebpftelemetry.NewEBPFErrorsCollector(); ec != nil {
+				telemetry.RegisterCollector(ec)
+			}
```
Comment (brycekahle) [inline cmd/system-probe/subcommands/run/command.go:347]: the lock contention collector is registered above
```suggestion
			if ec := ebpftelemetry.NewEBPFErrorsCollector(); ec != nil {
				telemetry.RegisterCollector(ec)
			}
```

**Review by brycekahle — CHANGES_REQUESTED** [review]

File: cmd/system-probe/subcommands/run/command.go (hunk 2 above)
Comment (val06) [reply cmd/system-probe/subcommands/run/command.go:347]: bad conflicts resolving :(

**Review by brycekahle — APPROVED** [review]

**Comment by val06** [comment]
I will wait for this [PR](https://github.com/DataDog/datadog-agent/pull/26849) and rebase on main afterwards

**Pull request merged**